Sources/TuistCache/ContentHashing/PlistContentHasher.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import TuistCore
+import TuistGraph
+
+public protocol PlistContentHashing {
+    func hash<T: PlistTypesProtocol>(plist: T) throws -> String
+}
+
+/// `PlistContentHasher`
+/// is responsible for computing a hash that uniquely identifies a property-list file (e.g. `Info.plist` or `.entitlements`)
+public final class PlistContentHasher: PlistContentHashing {
+    private let contentHasher: ContentHashing
+    // MARK: - Init
+
+    public init(contentHasher: ContentHashing) {
+        self.contentHasher = contentHasher
+    }
+
+    // MARK: - PlistContentHashing
+
+    public func hash<T: PlistTypesProtocol>(plist: T) throws -> String {
+        // TODO: DRY, improve generalization
+        if let plist = plist as? InfoPlist {
+            switch plist {
+            case let .file(path):
+                return try contentHasher.hash(path: path)
+            case let .dictionary(dictionary), let .extendingDefault(dictionary):
+                var dictionaryString: String = ""
+                for key in dictionary.keys.sorted() {
+                    let value = dictionary[key, default: "nil"]
```

Comment:
The logic here assumes that the data structure is only one-level deep. If the value under a given key is an array or a dictionary we should continue traversing.

Reply:
I'm just reusing the previous implementation: https://github.com/tuist/tuist/blob/main/Sources/TuistCache/ContentHashing/InfoPlistContentHasher.swift#L26-L32

If I'm understanding you correctly, this snipped should not work? 
````swift
"NSAppTransportSecurity": [
    "NSAllowsArbitraryLoads": true,
    "NSExceptionDomains": .dictionary(exceptionDomains),
],
````
This works on my side, on main and on this branch.